lib/logsource.c

```diff
@@ -38,6 +38,10 @@ gboolean accurate_nanosleep = FALSE;
 void
 log_source_wakeup(LogSource *self)
 {
+  if (!window_size_counter_suspended(&self->window_size))
```

Comment:
I would not add this condition to `log_source_wakeup`.
The name of this function tells me that it will wake the source up without condition.

Reply:
well... someone should check this suspend bit...
This is not the best and not the final place (this is why I added the wip flag :) )

Reply:
Oops, okay, sorry. I'll check it.

Reply:
I'm also playing with this a little bit.